internal/backend/user.go

```diff
@@ -37,22 +38,33 @@ type user struct {
 
 	// statesWG is
 	statesWG sync.WaitGroup
+
+	messageIDCounter uint64
```

Comment:
Since this is used with atomic instructions later on, it is better to leave this type as is to avoid issues with temporary addresses being incremented rather that the actual address.